agave_app/Controls.h

```diff
@@ -20,53 +20,48 @@
 
 class DualIconButton : public QPushButton
 {
+  Q_OBJECT
+  Q_PROPERTY(int state READ getState)
+
 private:
-  QIcon icon1;
-  QIcon icon2;
   QString tooltip1;
   QString tooltip2;
   QString statustip1;
   QString statustip2;
   int state;
 
 public:
-  DualIconButton(const QIcon& icon1,
-                 const QIcon& icon2,
-                 const QString& tooltip1,
+  DualIconButton(const QString& tooltip1,
                  const QString& statustip1,
                  const QString& tooltip2,
                  const QString& statustip2,
                  QWidget* parent = nullptr)
-    : icon1(icon1)
-    , icon2(icon2)
-    , tooltip1(tooltip1)
+    : tooltip1(tooltip1)
     , tooltip2(tooltip2)
     , statustip1(statustip1)
     , statustip2(statustip2)
     , state(0)
-    , QPushButton(parent)
+    , QPushButton(QIcon(), "", parent)
   {
-
-    setIcon(icon1);
     setToolTip(tooltip1);
     setStatusTip(statustip1);
     connect(this, &QPushButton::clicked, this, &DualIconButton::toggleIcon);
   }
 
   void toggleIcon() { setState(1 - state); }
 
+  int getState() const { return state; }
   void setState(int s)
   {
     state = s;
     if (state == 1) {
-      setIcon(icon2);
       setToolTip(tooltip2);
       setStatusTip(statustip2);
     } else {
-      setIcon(icon1);
       setToolTip(tooltip1);
       setStatusTip(statustip1);
     }
+    style()->polish(this);

```

Comment:
this line nudges the Qt style update code to respond to a property change so that it can refresh the icon

Reply:
Super interesting that you're able to specify the icons by state in the stylesheet!